crates/extensions/c8y_mapper_ext/src/operations/mod.rs

```diff
@@ -375,185 +530,280 @@ fn get_smartrest_response_for_upload_result(
+
+        assert_eq!(sut.running_operations.len(), 0);
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn handle_should_panic_when_background_task_panics() {
+        // we're immediately dropping test's temporary directory, so we'll get an error that a
+        // directory for the operation could not be created
+        let TestHandle {
+            operation_handler: mut sut,
+            ..
+        } = setup_operation_handler();
+
+        let mqtt_schema = sut.context.mqtt_schema.clone();
+
+        let entity_topic_id = EntityTopicId::default_main_device();
+        let entity_target = EntityTarget {
+            topic_id: entity_topic_id.clone(),
+            external_id: EntityExternalId::from("anything"),
+            smartrest_publish_topic: Topic::new("anything").unwrap(),
+        };
+
+        // spawn an operation to see if it's successfully joined when it's completed.
+        // particular operation used is not important, because we want to test only the handler.
+        // it would be even better if we could define some inline operation so test could be shorter
+        // TODO(marcel): don't assume operation implementations when testing the handler
+        let config_snapshot_operation = ConfigSnapshotCmd {
+            target: entity_topic_id,
+            cmd_id: "config-snapshot-1".to_string(),
+            payload: ConfigSnapshotCmdPayload {
+                status: CommandStatus::Successful,
+                tedge_url: Some("asdf".to_string()),
+                config_type: "typeA".to_string(),
+                path: None,
+                log_path: None,
+            },
+        };
+
+        sut.handle(
+            entity_target.clone(),
+            config_snapshot_operation.command_message(&mqtt_schema),
         )
         .await;
+        assert_eq!(sut.running_operations.len(), 1);
 
-        // Validate if the supported operation file is created
-        assert!(ttd
-            .path()
-            .join("operations/c8y/test-device:device:child1/c8y_DownloadConfigFile")
-            .exists());
+        // give OperationHandler time to handle message
+        // TODO(marcel): remove sleeps
+        tokio::time::sleep(Duration::from_millis(50)).await;
 
-        // Sending an updated list of config types
-        mqtt.send(MqttMessage::new(
-            &Topic::new_unchecked("te/device/child1///cmd/config_update"),
-            r#"{"types" : [ "typeB", "typeC", "typeD" ]}"#,
-        ))
-        .await
-        .expect("Send failed");
-
-        // Assert that the updated config type list does not trigger a duplicate supported ops message
-        assert_received_contains_str(
-            &mut mqtt,
-            [(
-                "c8y/s/us/test-device:device:child1",
-                "119,typeB,typeC,typeD",
-            )],
+        // normally clearing message would be sent by operation task, but in this case we send it
+        // manually
```

Comment:
```suggestion
        // normally clearing message would be sent by operation task.
        // Using it here just as a dummy, to call `handle` with the same cmd-id, so that it panics
```